account_comment_template/tests/test_account_move_report.py

```diff
@@ -3,68 +3,55 @@
 # Copyright 2021-2022 Tecnativa - Víctor Martínez
 # License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
 
-from odoo.tests.common import Form, TransactionCase
+from odoo.tests import Form, tagged
 
+from odoo.addons.account.tests.common import AccountTestInvoicingCommon
 
-class TestAccountInvoiceReport(TransactionCase):
-    def setUp(self):
-        super().setUp()
-        self.company = self.env.ref("base.main_company")
-        self.base_comment_model = self.env["base.comment.template"]
-        self.before_comment = self._create_comment("before_lines")
-        self.after_comment = self._create_comment("after_lines")
-        self.partner = self.env["res.partner"].create({"name": "Partner Test"})
-        self.partner.base_comment_template_ids = [
-            (4, self.before_comment.id),
-            (4, self.after_comment.id),
+
+@tagged("post_install", "-at_install")
+class TestAccountInvoiceReport(AccountTestInvoicingCommon):
+    @classmethod
+    def setUpClass(cls, chart_template_ref=None):
+        super().setUpClass(chart_template_ref=chart_template_ref)
+        cls.company = cls.env.ref("base.main_company")
+        cls.base_comment_model = cls.env["base.comment.template"]
+        cls.move_obj = cls.env.ref("account.model_account_move")
+        cls.before_comment = cls._create_comment(cls, position="before_lines")
+        cls.after_comment = cls._create_comment(cls, position="after_lines")
+        cls.partner = cls.env["res.partner"].create({"name": "Partner Test"})
+        cls.partner.base_comment_template_ids = [
+            (4, cls.before_comment.id),
+            (4, cls.after_comment.id),
         ]
-        self.income_account = self.env["account.account"].search(
-            [
-                ("user_type_id.name", "=", "Income"),
-                ("company_id", "=", self.company.id),
-            ],
-            limit=1,
-        )
-        self.journal_sale = self.env["account.journal"].create(
-            {
-                "name": "Test journal sale",
-                "code": "TST-JRNL-S",
-                "type": "sale",
-                "company_id": self.company.id,
-            }
-        )
-        move_form = self._create_invoice()
-        self.invoice = move_form.save()
+        move_form = cls._create_invoice(cls)
```

Comment:
`AccountTestInvoicingCommon` has a helper method for creating an invoice. You can use it and save lines of code.